playwright/tests/page-objects/inscription.ts

```diff
@@ -235,46 +151,82 @@ class Select {
 
 export class Input {
   readonly locator: Locator;
+  readonly outputLocator: Locator;
+  readonly inputLocator: Locator;
 
   constructor(
     readonly page: Page,
     readonly parentLocator: Locator,
-    options?: { label?: string; nth?: number; type?: string }
+    options?: { label?: string; nth?: number; type?: 'number' | 'id' }
   ) {
     const role = options?.type === 'number' ? 'spinbutton' : 'textbox';
-    if (options?.label) {
-      this.locator = parentLocator.getByRole(role, { name: options.label }).first();
+    const badgeLocator = parentLocator.locator('.badge-field');
+    if (role === 'spinbutton' || options?.type === 'id') {
+      if (options?.label) {
+        this.locator = parentLocator.getByRole(role, { name: options.label }).first();
+      } else {
+        this.locator = parentLocator.getByRole(role).nth(options?.nth ?? 0);
+      }
+      this.inputLocator = this.locator;
+    } else if (options?.label) {
+      this.locator = badgeLocator.filter({ has: page.locator('.ui-fieldset-label', { hasText: options.label }) }).first();
+      this.inputLocator = this.locator.getByRole(role, { name: options.label }).first();
+      this.outputLocator = this.locator.locator('output').first();
     } else {
-      this.locator = parentLocator.getByRole(role).nth(options?.nth ?? 0);
+      this.locator = badgeLocator.locator('.badge-field').nth(options?.nth ?? 0);
+      this.inputLocator = this.locator.getByRole(role).first();
+      this.outputLocator = this.locator.locator('output').first();
+    }
+  }
+
+  async focus() {
+    if (!(await this.inputLocator.isVisible())) {
+      await this.locator.click();
+    }
+  }
+
+  async focusOut() {
+    if (await this.inputLocator.isVisible()) {
+      await this.inputLocator.blur();
     }
   }
 
   async clear() {
-    await this.locator.clear();
+    this.focus();
+    await this.inputLocator.clear();
   }
 
   async fill(value: string) {
+    await this.focus();
     await this.clear();
-    await this.locator.fill(value);
+    await this.inputLocator.fill(value);
   }
 
   async openBrowser() {
-    await this.locator.locator('+ .ui-button').click();
+    await this.focus();
+    await this.inputLocator.locator('+ .ui-button').click();
     const browser = new Browser(this.page);
     await expect(browser.view).toBeVisible();
     return browser;
   }
 
   async expectValue(value: string | RegExp) {
```

Comment:
would also be fine if assert the badge output
or maybe have a second method one for the input value, one for the badge output